java/vector/src/main/java/org/apache/arrow/vector/complex/RunEndEncodedVector.java

```
@@ -0,0 +1,683 @@
+   * @param runEndsVector The vector represents run ends. Only Zero vector or type int vector with
+   *     size 16, 32 is allowed
+   * @param valuesVector The vector represents values
+   * @param callBack A schema change callback.
+   */
+  public RunEndEncodedVector(
+      Field field,
+      BufferAllocator allocator,
+      FieldVector runEndsVector,
+      FieldVector valuesVector,
+      CallBack callBack) {
+    super(allocator);
+    this.field = field;
+    this.callBack = callBack;
+    this.valueCount = 0;
+    this.runEndsVector = runEndsVector;
+    this.valuesVector = valuesVector;
+  }
+
+  /** ValueVector interface */
+
+  /**
+   * Allocate new buffers. ValueVector implements logic to determine how much to allocate.
+   *
+   * @throws OutOfMemoryException Thrown if no memory can be allocated.
+   */
+  @Override
+  public void allocateNew() throws OutOfMemoryException {
+    if (!allocateNewSafe()) {
+      throw new OutOfMemoryException("Failure while allocating memory");
+    }
+  }
+
+  /**
+   * Allocates new buffers. ValueVector implements logic to determine how much to allocate.
+   *
+   * @return Returns true if allocation was successful.
+   */
+  @Override
+  public boolean allocateNewSafe() {
+    initializeChildrenFromFields(field.getChildren());
+    for (FieldVector v : getChildrenFromFields()) {
+      boolean isAllocated = v.allocateNewSafe();
+      if (!isAllocated) {
+        v.clear();
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Allocate new buffer with double capacity, and copy data into the new buffer. Replace vector's
+   * buffer with new buffer, and release old one
+   */
+  @Override
+  public void reAlloc() {
+    for (FieldVector v : getChildrenFromFields()) {
+      v.reAlloc();
+    }
+  }
+
+  @Override
+  public BufferAllocator getAllocator() {
+    return allocator;
+  }
+
+  @Override
+  protected FieldReader getReaderImpl() {
+    return null;
```

nit: shall we throw unsupported here?